File: c/125_pattern_17.c
#include<stdio.h>
int main() {
    int n;
    printf("Enter n :- ");
    scanf("%d",&n);
    for(int i=0;i<n;i++) {
        for(int j=0;j<n;j++) {
            if(i==0 || i==n-1 || i==n/2) {
                printf("*");
            }
            else {
                if(i<n/2 && j==0) printf("*");
                else if(i>n/2 && j==n-1) printf("*");
                else printf(" ");
            }
        }
        printf("\n");
    }
}

/*

Output :-
Enter n :- 5
*****
*
*****
    *
*****

*/

File: python/code/125_pattern17.py
n = int(input("Enter n :- "))
for i in range(n):
    for j in range(n):
        if(i == 0 or i == n-1 or i == n//2):
            print("*", end='')
        else:
            if(i < n//2 and j == 0):
                print("*", end='')
            elif(i > n//2 and j == n-1):
                print("*", end='')
            else:
                print(" ", end='')
    print()
    
# Output :-
# Enter n :- 5
# *****
# *
# *****
#     *
# *****